### PR TITLE
Georg/ch6977/esb race condition detected when running

### PR DESCRIPTION
**Pull request by msiebeling**
I tested this PR by running the following unit tests:
- pkcs7 unit tests
- sds unit tests (after updating vendoring)
- eg036-export-binary unit tests with option --race (after updating vendoring) 
- export-service (after including above eg036-export-binary)

### CODE REVIEW

File: ber.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"errors"
 )
 
-var encodeIndent = 0
+// var encodeIndent = 0
```
Comment (tpjg) [inline ber.go:8]: Obviously - commented out code is not good ... but I see it's all from upstream, so it's okay.

File: pkcs7.go [hunk 2]
```diff
@@ -85,7 +85,7 @@ type recipientInfo struct {
 type encryptedContentInfo struct {
 	ContentType                asn1.ObjectIdentifier
 	ContentEncryptionAlgorithm pkix.AlgorithmIdentifier
-	EncryptedContent           asn1.RawValue `asn1:"tag:0,optional,explicit"`
+	EncryptedContent           asn1.RawValue `asn1:"tag:0,optional"`
```
Comment (tpjg) [inline pkcs7.go:88]: The merges look good to me, I see it's all from upstream. I do have one-and-a-half suggestions, basically the only difference with upstream code is around line 171, the upstream version is below.

Georg made the changes as it looked the original upstream did not work on large files, hence the looping until everything is parsed and appending with `content = append(content, parsedContent...)`.
I suggest you make it explicit, maybe with comments in the code, that this is where we differ from upstream. And maybe, just maybe, the normal upstream version now works on bigger files (although I didn't check if relevant changes were made to asn1.Unmarshal)?

```	
// Compound octet string
	if compound.IsCompound {
		if _, err = asn1.Unmarshal(compound.Bytes, &content); err != nil {
			return nil, err
		}
	} else {
```

**Review by tpjg — APPROVED** [review]
I review and I'm pretty sure it's safe to merge (and I know @msiebeling tested with SDS and so on). However I suggest adding a comment around the piece where our code differs from upstream just for clarity.

File: pkcs7.go (hunk 2 above)
Comment (msiebeling) [reply pkcs7.go:88]: Ok, i will add a comment to mark the change we did compared to upstream.
I have also testing with the upstream code without the changes Georg made but than some unit tests are failing. So this change is required.

**Pull request merged**